src/Test/PHPat.php

```diff
@@ -10,6 +10,11 @@ class PHPat
 {
     public static function rule(): SubjectStep
     {
-        return new SubjectStep(new RelationRule());
+        // get architecture rule name from stack trace ( e.g. test_xxxx )
+        $ruleName = debug_backtrace()[1]['function'] ?? '';
+
+        $rule           = new RelationRule();
+        $rule->ruleName = $ruleName;
+        return new SubjectStep($rule);
```

Comment:
In #208, the method name in TestParser was carried.
https://github.com/carlosas/phpat/blob/c0036b94e530b6c55c6f7f3b31b67f817ed8f408/src/Test/TestParser.php#L56-L61

However, if ruleName is a member variable, the same acquisition method cannot be used.
The problem is that RuleBuilder's rule is a protected variable.
https://github.com/carlosas/phpat/blob/29660717987f3c54c6ea54447a10d9931f5fde71/src/Test/Builder/AbstractStep.php#L11

So I tried to solve it by picking up the method name of the caller when initializing the Rule.
The assumption is that the return PHPat::rule() statement must be used.

Reply:
... Perhaps adding a public setRuleName() to RuleBuilder would help?

Reply:
fixed.